ownership-chain/runtime/src/tests/xcm_tests.rs

```diff
@@ -0,0 +1,165 @@
+#[test]
+fn basic_dmp() {
+	MockNet::reset();
+
+	let remark = parachain::RuntimeCall::System(
+		frame_system::Call::<parachain::Runtime>::remark_with_event { remark: vec![1, 2, 3] },
+	);
+
+	Relay::execute_with(|| {
+		assert_ok!(RelayChainPalletXcm::send_xcm(
+			Here,
+			Parachain(1),
+			Xcm(vec![
+				WithdrawAsset(
+					vec![MultiAsset {
+						id: MultiLocation::here().into(),
+						fun: cumulus_primitives_core::Fungibility::Fungible(UNIT)
+					}]
+					.into()
+				),
+				BuyExecution {
+					fees: MultiAsset {
+						id: MultiLocation::here().into(),
+						fun: cumulus_primitives_core::Fungibility::Fungible(UNIT)
+					},
+					weight_limit: Unlimited,
+				},
+				Transact {
+					origin_kind: OriginKind::SovereignAccount,
+					require_weight_at_most: Weight::from_parts(1_000_000_000, 1024 * 1024),
+					call: remark.encode().into(),
+				}
+			]),
+		));
+	});
+
+	// Execute remote transact and verify that `Remarked` event is emitted.
+	ParaA::execute_with(|| {
+		use parachain::{RuntimeEvent, System};
+		assert!(System::events().iter().any(|r| matches!(
+			r.event,
+			RuntimeEvent::System(frame_system::Event::Remarked { .. })
+		)));
+	});
+}
+
+#[test]
+fn para_to_para_native_transfer_and_back() {
+	MockNet::reset();
+
+	// in ParaB, we need to set up and register the token of ParaA
+	ParaB::execute_with(|| {
+		// we only need to create the asset, registering location and setting unit price
+		// per second is not necessary, since it is already mocked by
+		assert_ok!(Assets::force_create(
+			parachain::RuntimeOrigin::root(),
+			2_u32,
+			ALITH,
+			false,
+			UNIT
+		));
+	});
+
+	let amount = 100 * UNIT;
+	ParaA::execute_with(|| {
+		// bob in ParaB
+		let destination = MultiLocation {
+			parents: 1,
+			interior: X2(
+				Parachain(2),
+				AccountKey20 { network: Some(NetworkId::Kusama), key: BOBTH.0 },
+			),
+		};
+
+		assert_ok!(ParachainXtokens::transfer(
+			parachain::RuntimeOrigin::signed(ALITH),
+			0,
+			amount,
+			Box::new(destination.into()),
+			Unlimited,
+		));
+	});
+
+	ParaB::execute_with(|| {
+		use parachain::{RuntimeEvent, System};
+
+		assert!(System::events().iter().any(|r| {
+			matches!(r.event, RuntimeEvent::MsgQueue(mock_msg_queue::Event::Success(_)))
+		}));
+
+		assert!(System::events().iter().any(|r| {
+			matches!(r.event, RuntimeEvent::Assets(pallet_assets::Event::Issued { .. }))
+		}));
+
+		// try to round up the amount to the nearest second, since we don't know how much fee was
+		// charged
+		let rounded_balance = Assets::total_balance(2, &BOBTH) / UNIT * UNIT;
+
+		// we have 1 UNIT of tolerance, i.e at most 1 UNIT of was charged
+		assert_eq!(rounded_balance, amount - UNIT);
+
+		// now transfer back some of the tokens
+		let destination = MultiLocation {
+			parents: 1,
+			interior: X2(
+				Parachain(1),
+				AccountKey20 { network: Some(NetworkId::Kusama), key: ALITH.0 },
+			),
+		};
+
+		assert_ok!(ParachainXtokens::transfer_multiasset(
```

Comment:
why `transfer_multiasset`? would be a `withdraw` instead?

Reply:
`transfer` uses native currency, `transfer_multiasset` uses a generic asset type, i.e foreign assets that can be paid as tx fee + native currencies

Reply:
but because we are transfer back the tokens wouldn't be closer to the reality use `withdraw`? this is the way xtokens does: https://github.com/open-web3-stack/open-runtime-module-library/blob/polkadot-v0.9.43/xtokens/src/lib.rs#L705

Reply:
there is no `withdraw` extrinsic, wdym?

Reply:
You are right I had forgotten that `withdraw` is part of `xtokens` pallet, but now it's a bit unclear for me. How are we going to transfer back tokens? By making `transfer_multiasset` we will leave CLD as a reserve transfer in origin parachain (Astar) and we don't want that. I'm rethinking that maybe we do need `xtokens` in order to perform the correct transferring back which is by using `withdraw`

Reply:
you might have confused the difference:

- `XTokens::transfer_multiasset` does use `WITHDRAW_ASSET` instrcution
- `PolkadotXcm::limited_reserve_asset_transfer` does NOT use `WITHDRAW_ASSET` instruction

so, here, when I am doing `XTokens::transfer_multiasset` from `ParaB`, it is withdrawing the asset from `ParaB`

Reply:
omg you are totally right sorry, I should have seen the `XTokens::transfer_multiasset` deeper, in that case could we add a test that checks that Sovereign Account of ParaA in ParaB has 0 tokens? just to be sure, do you think is worth it?

Reply:
sure, I can

Reply:
I will just assert that total issuance of ParaA token in ParaB is 0, meaning everything was withdrawn